Mathlib/CategoryTheory/Bicategory/Functor/Strict.lean

```diff
@@ -0,0 +1,146 @@
+    F.mapComp'_hom_comp_whiskerLeft_mapComp'_hom_assoc _ _ _ _ _ _ h₀₂ h₁₃ hf]
+  simp
+
+@[reassoc]
+lemma whiskerLeft_mapComp'_inv_comp_mapComp'_inv (hf : f₀₁ ≫ f₁₃ = f) :
+    F.map f₀₁ ◁ (F.mapComp' f₁₂ f₂₃ f₁₃ h₁₃).inv ≫ (F.mapComp' f₀₁ f₁₃ f hf).inv =
+    (α_ _ _ _).inv ≫ (F.mapComp' f₀₁ f₁₂ f₀₂ h₀₂).inv ▷ F.map f₂₃ ≫
+      (F.mapComp' f₀₂ f₂₃ f).inv := by
+  simp [← cancel_mono (F.mapComp' f₀₂ f₂₃ f).hom,
+    F.mapComp'_inv_comp_mapComp'_hom _ _ _ _ _ _ h₀₂ h₁₃ hf]
+
+@[reassoc]
+lemma mapComp'_hom_comp_mapComp'_hom_whiskerRight (hf : f₀₂ ≫ f₂₃ = f) :
+    (F.mapComp' f₀₂ f₂₃ f).hom ≫ (F.mapComp' f₀₁ f₁₂ f₀₂ h₀₂).hom ▷ F.map f₂₃ =
+    (F.mapComp' f₀₁ f₁₃ f).hom ≫ F.map f₀₁ ◁ (F.mapComp' f₁₂ f₂₃ f₁₃ h₁₃).hom ≫
+      (α_ _ _ _).inv := by
+  rw [F.mapComp'_hom_comp_whiskerLeft_mapComp'_hom_assoc _ _ _ _ _ f h₀₂ h₁₃ (by aesop_cat)]
+  simp
+
+@[reassoc]
+lemma mapComp'_inv_whiskerRight_comp_mapComp'_inv (hf : f₀₂ ≫ f₂₃ = f) :
+    (F.mapComp' f₀₁ f₁₂ f₀₂ h₀₂).inv ▷ F.map f₂₃ ≫ (F.mapComp' f₀₂ f₂₃ f).inv =
+    (α_ _ _ _).hom ≫ F.map f₀₁ ◁ (F.mapComp' f₁₂ f₂₃ f₁₃ h₁₃).inv ≫
+      (F.mapComp' f₀₁ f₁₃ f).inv := by
+  rw [whiskerLeft_mapComp'_inv_comp_mapComp'_inv _ _ _ _ _ _ f h₀₂ h₁₃,
+    Iso.hom_inv_id_assoc]
+
+end associativity
+
+section CommSq
+
+variable {X₁ X₂ Y₁ Y₂ Z₁ Z₂ : B}
+
+section
+
+variable {t : X₁ ⟶ Y₁} {l : X₁ ⟶ X₂} {r : Y₁ ⟶ Y₂} {b : X₂ ⟶ Y₂} (sq : CommSq t l r b)
+
+/-- Given a commutative square `CommSq t l r b` in a strict bicategory `B` and
+a pseudofunctor from `B`, this is the natural isomorphism
+`F.map t ≫ F.map r ≅ F.map l ≫ F.map b`. -/
+def isoMapOfCommSq : F.map t ≫ F.map r ≅ F.map l ≫ F.map b :=
+  (F.mapComp t r).symm ≪≫ F.mapComp' _ _ _ (by rw [sq.w])
+
+lemma isoMapOfCommSq_eq (φ : X₁ ⟶ Y₂) (hφ : t ≫ r = φ) :
+    F.isoMapOfCommSq sq = (F.mapComp' t r φ (by rw [hφ])).symm ≪≫
+      F.mapComp' l b φ (by rw [← hφ, sq.w]) := by
+  subst hφ
+  simp [isoMapOfCommSq, mapComp'_eq_mapComp]
+
+end
+
+/-- Equational lemma for `Pseudofunctor.isoMapOfCommSq` when
+both vertical maps of the square are the same and horizontal maps are identities. -/
+lemma isoMapOfCommSq_horiz_id (f : X₁ ⟶ X₂) :
+    F.isoMapOfCommSq (t := 𝟙 _) (l := f) (r := f) (b := 𝟙 _) ⟨by simp⟩ =
+      whiskerRightIso (F.mapId X₁) (F.map f) ≪≫ λ_ _ ≪≫ (ρ_ _).symm ≪≫
+        (whiskerLeftIso (F.map f) (F.mapId X₂)).symm := by
+  ext
+  rw [isoMapOfCommSq_eq _ _ f (by simp), mapComp'_comp_id, mapComp'_id_comp]
+  simp
+
+/-- Equational lemma for `Pseudofunctor.isoMapOfCommSq` when
+both horizontal maps of the square are the same and vertical maps are identities. -/
+lemma isoMapOfCommSq_vert_id (f : X₁ ⟶ X₂) :
+    F.isoMapOfCommSq (t := f) (l := 𝟙 _) (r := 𝟙 _) (b := f) ⟨by simp⟩ =
+      whiskerLeftIso (F.map f) (F.mapId X₂) ≪≫ ρ_ _ ≪≫ (λ_ _).symm ≪≫
+        (whiskerRightIso (F.mapId X₁) (F.map f)).symm := by
+  ext
+  rw [isoMapOfCommSq_eq _ _ f (by simp), mapComp'_comp_id, mapComp'_id_comp]
+  simp
```

Comment:
What's the use case for these two lemmas?

Reply:
These are basic API lemmas for `isoMapOfCommSq`. Soon, I will add similar lemmas for the horizontal and vertical composition of squares.
I am using it there https://github.com/joelriou/topcat-model-category/blob/4b4b46eda5c6038efa64d7c70e9c46cfb6e5b6dd/TopCatModelCategory/PseudoFunctor/MorphismProperty.lean#L75-L103 and https://github.com/joelriou/topcat-model-category/blob/4b4b46eda5c6038efa64d7c70e9c46cfb6e5b6dd/TopCatModelCategory/SSet/KanComplexWRetracts.lean in order to prove that if a map of (Kan) simplicial sets induces an equivalence of categories on the fundamental groupoid, then the same applies to any retract. It will also be used when studying base change theorems (e.g. the base change theorem holds when horizontal or vertical maps are identities).